spec/custom_facts/core/execution_spec.rb

```diff
@@ -44,4 +44,50 @@
     expect(impl).to receive(:execute).with('waffles', {})
     execution.execute('waffles')
   end
+
+  context 'when running an actual command' do
+    before do
+      allow(impl).to receive(:which).with('waffles').and_return('/under/the/honey/are/the/waffles')
+    end
+
+    context 'with default parameters' do
+      it 'execute the found command' do
+        expect(impl).to receive(:execute_command).with('/under/the/honey/are/the/waffles', :raise, nil, nil)
+        execution.execute('waffles')
+      end
+    end
+
+    context 'with a timeout' do
+      it 'execute the found command with a timeout' do
+        expect(impl).to receive(:execute_command).with('/under/the/honey/are/the/waffles', :raise, nil, 90)
+        execution.execute('waffles', timeout: 90)
+      end
+    end
+
+    context 'with a deprecated time_limit' do
+      it 'execute the found command with a timeout' do
+        expect(impl).to receive(:execute_command).with('/under/the/honey/are/the/waffles', :raise, nil, 90)
+        execution.execute('waffles', time_limit: 90)
+      end
+
+      it 'emits a warning' do
+        expect_any_instance_of(Facter::Log).to receive(:warn)
```

Comment:
I think you can do something similar to https://github.com/puppetlabs/facter/blob/main/spec/facter/resolvers/windows/memory_spec.rb to avoid `expect_any_instance_of`.

Reply:
I tried, unfortunately it does not work.

I feel like the logger is initialized when the module `Facter::Core::Execution` is evaluated (`lib/facter/custom_facts/core/execution.rb` is loaded, `@@impl` is initialized with a `Facter::Core::Execution::Base` derived class, the Logger is instanciated and the module loading continue).

I was pushing this code hopping someone could help :wink:

Reply:
@GabrielNagy I could workaround this issue by getting the logger differently:

```
let(:logger) { impl.instance_variable_get(:@log) }
```

Not very clean, but cleaner I guess since we have "weird" thing in only one place.